src/components/NewArticle/NewArticle.tsx

```diff
@@ -0,0 +1,259 @@
+    }
+  ),
+}
+
+const Input = dynamic<InputProps>(
+  () => import('@/components/Input/Input').then((module) => module.Input),
+  { ssr: false }
+)
+
+const Wrapper = styled.form`
+  display: flex;
+  flex-direction: column;
+  row-gap: ${({ theme }) => theme.spacings.xmedium};
+  width: 100%;
+  max-width: 700px;
+`
+
+const Title = styled.h1`
+  font-size: ${({ theme }) => theme.fonts.sizes.huge};
+  color: ${({ theme }) => theme.colors.black[400]};
+`
+
+const FieldWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  row-gap: ${({ theme }) => theme.spacings.xxsmall};
+`
+
+const PublishBtn = styled(Button)`
+  align-self: flex-end;
+`
+
+const AlertContentWrapper = styled.div`
+  width: 100%;
+  display: flex;
+  gap: ${({ theme }) => theme.spacings.xsmall};
+`
+
+const newArticleSchema = object({
+  title: string()
+    .required('Title is a required field and cannot be empty')
+    .max(32, 'Maximum 32 characters required!'),
+  description: string()
+    .required('Description is a required field and cannot be empty')
+    .max(64, 'Maximum 64 characters required!'),
+  body: string().required('Body is a required field and cannot be empty!'),
+})
+
+type NewArticleFieldValues = {
+  title: string
+  description: string
+  body: string
+}
+
+export type NewArticleRequest = {
+  article: NewArticleFieldValues
+}
+
+const initialFieldValues: NewArticleFieldValues = {
+  title: '',
+  description: '',
+  body: '',
+}
+
+const storageKey = 'new-article-editor'
+
+const getStorage = () => {
+  if (typeof window === 'undefined') return initialFieldValues
+
+  const cachedStorage = fromNullable(localStorage.getItem(storageKey))
+
+  return isSome(cachedStorage)
+    ? superJsonParse<NewArticleFieldValues>(cachedStorage.value)
+    : initialFieldValues
+}
+
+const useLocalStorage = () => {
+  const [storage, setStorage] = useState<NewArticleFieldValues>(getStorage)
+
+  useEffect(
+    () => localStorage.setItem(storageKey, superJsonStringify(storage)),
+    [storage]
+  )
+
+  return { storage, setStorage }
+}
+
+const firstInputRef = (node: HTMLInputElement | null) =>
+  node ? node.focus() : null
+
+export const NewArticle = () => {
+  const { mutate: createArticle, data, error, isLoading } = useNewArticle()
+  const { storage, setStorage } = useLocalStorage()
+
+  const handleSubmit = (values: NewArticleFieldValues) => {
+    const newCreateArticleValues: NewArticleRequest = {
+      article: values,
+    }
+    createArticle(newCreateArticleValues)
+  }
+
+  const formik = useFormik<NewArticleFieldValues>({
+    initialValues: storage,
+    validationSchema: newArticleSchema,
+    validateOnChange: false,
+    validateOnBlur: false,
+    onSubmit: (values) => handleSubmit(values),
+  })
+
+  useEffect(() => {
+    const updatedFieldValues: NewArticleFieldValues = {
+      title: formik.values.title,
+      description: formik.values.description,
+      body: formik.values.body,
+    }
+    setStorage(updatedFieldValues)
```

Comment:
setStorage(formik.values)

Reply:
Fixed in this [commit](https://github.com/joaooliveiradev/conduit-frontend/pull/43/commits/6479fa461658d52d71ce7613a126c392e06e6ed5).